component/src/main/java/org/wso2/extension/siddhi/execution/tensorflow/util/CoreUtils.java

```diff
@@ -0,0 +1,324 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.execution.tensorflow.util;
+
+import org.tensorflow.DataType;
+import org.tensorflow.SavedModelBundle;
+import org.tensorflow.Shape;
+import org.tensorflow.Tensor;
+import org.tensorflow.framework.SignatureDef;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.executor.VariableExpressionExecutor;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.nio.ByteBuffer;
+import java.nio.DoubleBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Stack;
+
+/**
+ * Core util functions for TensorFlow SP extension
+ */
+public class CoreUtils {
+
+    public static VariableExpressionExecutor[] extractAndValidateTensorFlowInputs(
+            ExpressionExecutor[] attributeExpressionExecutors, int startIndex, int noOfInputs) {
+
+        VariableExpressionExecutor[] inputVariableExpressionExecutors = new VariableExpressionExecutor[noOfInputs];
+
+        for (int i = startIndex; i < startIndex + noOfInputs; i++) {
+            if (attributeExpressionExecutors[i] instanceof  VariableExpressionExecutor) {
+                inputVariableExpressionExecutors[i - startIndex] = ((VariableExpressionExecutor)
+                        attributeExpressionExecutors[i]);
+            } else {
+                throw new SiddhiAppValidationException("The parameter at index " + (i + 1) + " is not a variable " +
+                        "attribute (VariableExpressionExecutor) present in the stream definition. Found " +
+                        attributeExpressionExecutors[i].getClass().getCanonicalName());
+            }
+        }
+        return inputVariableExpressionExecutors;
+    }
+
+    public static List<Attribute> getReturnAttributeList(SignatureDef signatureDef, int noOfOutputs,
+                                                         SavedModelBundle tensorFlowSavedModel,
+                                                         String[] outputNamesArray) {
+        List<Attribute> attributeList = new ArrayList<>(noOfOutputs);
+        for (int i = 0; i < noOfOutputs; i++) {
+
```

Comment:
@niruhan remove the unnecessary spaces. check the other files as well.

Reply:
Fixed in https://github.com/siddhi-io/siddhi-execution-tensorflow/pull/20